src/connection.js

```diff
@@ -550,8 +552,16 @@ class Connection extends EventEmitter {
       this.socket.on('close', this.socketClose);
       this.socket.on('end', this.socketEnd);
       this.messageIo = new MessageIO(this.socket, this.config.options.packetSize, this.debug);
-      this.messageIo.on('data', (data) => { this.dispatchEvent('data', data); });
-      this.messageIo.on('message', () => { this.dispatchEvent('message'); });
+      this.messageIo.incomingMessageStream.on('data', (message) => {
+        if (this.state.name == 'SentClientRequest') {
```

Comment:
This needs to be cleaned up. 👍 

According to section `2.2.4 - Packet Data Token and Tokenless Data Streams` of the TDS spec, I believe the decision on whether the message contents is a token stream or not should not be based on the current state of the connection, but on the message type instead.

Piping directly into `this.tokenStreamParser.parser` is not great from a coupling perspective, but the token stream parser is a bit of a mess right now and needs to be cleaned up anyway, so I'm willing to compromise here.

> A cleaner option may be to capture this logic in a helper method, let this flow through dispatch event and invoke the helper in the 'data' methods of the respective states.

This is what I want to get away from here. 😄 I'll see if I can clean this up.